packages/shared/src/components/streak/popup/ReadingStreakPopup.tsx

```diff
@@ -19,52 +23,74 @@ const streakDays = [
   addDays(today, 4),
 ]; // these dates will then be compared to the user's post views
 
-export function ReadingStreakPopup(): ReactElement {
-  return (
-    <div className="flex flex-col">
-      <div className="flex flex-row">
-        <StreakSection streak={44} label="Current streak" />
-        <StreakSection streak={100} label="Longest streak 🏆" />
-      </div>
-      <div className="mt-6 flex flex-row gap-2">
-        {streakDays.map((value) => {
-          const day = value.getDay();
-          const date = value.getDate();
+interface ReadingStreakPopupProps {
+  streak: UserStreak;
+}
+
+export function ReadingStreakPopup({
+  streak,
+}: ReadingStreakPopupProps): ReactElement {
+  const { user } = useAuthContext();
+  const { data: history } = useQuery(
+    generateQueryKey(RequestKey.ReadingStreak30Days, user),
+    () => getReadingStreak30Days(user.id),
+  );
 
-          // this is just a dummy logic to simulate the popup with different values
-          const getStreak = () => {
-            if (date === dateToday) {
-              return Streak.Pending;
-            }
+  const streaks = useMemo(() => {
+    return streakDays.map((value) => {
+      const day = value.getDay();
+      const date = value.getDate();
+      const isFuture = value > today;
+      const isCompleted =
+        !isFuture &&
+        history?.some(({ date: historyDate, reads }) => {
+          const dateToCompare = new Date(historyDate);
+          const sameDate = isSameDay(dateToCompare, value);
 
-            if (day === 0 || day === 6) {
-              return Streak.Freeze;
-            }
+          return sameDate && reads > 0;
+        });
 
-            if (dateToday > date) {
-              return Streak.Completed;
-            }
+      const getStreak = () => {
+        if (isCompleted) {
+          return Streak.Completed;
+        }
 
-            return Streak.Upcoming;
-          };
+        if (day === 0 || day === 6) {
```

Comment:
should this not be 5 & 6?
```suggestion
        if (day === 5 || day === 6) {
```

Reply:
0 is Sunday, 6 is Saturday, [see the `getDay` function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDay#return_value).

Maybe we can add const or enum values for these here as well, like you did in the other PR @sshanzel ?

Reply:
6 and 0 is Saturday and Sunday. Unfortunately, Sunday starts at `0` in JS 😅

Reply:
> Maybe we can add const or enum values for these here as well, like you did in the other PR @sshanzel ?

Yeah, makes sense. Let's introduce it here as well.